Fixed #7497 -- Allowed overriding the order of apps and models in admin. (#15483)
django/django: docs/ref/contrib/admin/index.txt

``AdminSite`` methods
---------------------

> .. method:: AdminSite.get_app_list(request, app_label=None)

Reviewer: `AdminSite` methods are alphabetized, please move it below `each_context()` docs.

>     .. versionadded:: 4.1

Reviewer: This method exists for a long time, so I would remove `versionadded` annotation and only add `versionchanged` about `app_label` argument:
```
.. method:: AdminSite.get_app_list(request, app_label=None)

    ...

    .. versionchanged:: 4.1

        The ``app_label`` argument was added.

.. method:: AdminSite.has_permission(request) 
```

    Returns a list of applications from the :doc:`application registry
    </ref/applications/>` available for the current user. Each entry in the

>     list is a dict representing an application with the following keys:

Reviewer: ```suggestion
    list is a dictionary representing an application with the following keys:
```

Reviewer: No need, we try to avoid trivial wording changes unless it's a big problem as it creates extra work for documentation translators.

    * ``app_label``: the application label
    * ``app_url``: the URL of the application index in the admin
    * ``has_module_perms``: a boolean indicating if displaying and accessing of
      the module's index page is permitted for the current user
    * ``models``: a list of the models available in the application
    * ``name``: name of the application

    Each model is a dict with the following keys:

    * ``model``: the model class
    * ``object_name``: class name of the model
    * ``name``: plural name of the model
    * ``perms``: a ``dict`` tracking ``add``, ``change``, ``delete``, and
      ``view`` permissions
    * ``admin_url``: admin changelist URL for the model
    * ``add_url``: admin URL to add a new model instance

    List can be limited to a single application by providing ``app_label`` argument.

    List is sorted alphabetically by application's ``name`` key. Additionally

>     ``models`` list of each application is sorted by model's ``name`` key. If
>     non-alphabetical order makes more sense and you want to customize application
>     and/or model order in admin site, you can override this method in a custom
>     AdminSite class.

Reviewer: Maybe:
```suggestion
    ``models`` list of each application is sorted by model's ``name`` key.
    
    You can override this method to change the default ordering of apps and
    models in the admin.
```

    .. warning::

        Do not override AdminSite.get_app_list method for access control purposes.

> 

Reviewer: IMO this warning is unnecessary.

Author: Attack vector I had in mind was that somebody abused get_app_list to hide apps and models, but actually left admin pages pretty much accessible by direct links. I'm kinda paranoid about security, seen people do unbelievable crazy stuff :-(
